auth/utils/read-external-sources.ts

```diff
@@ -97,8 +96,31 @@ function filterPropertiesByServiceName(envObj: any, serviceName: string): any {
  * Pulls credentials from VCAP_SERVICES env property that IBM Cloud sets
  *
  */
-function getCredentialsFromCloud(serviceName: string): any {
-  const credentials = vcapServices.getCredentials(serviceName);
+function getVCAPCredentialsForService(name) {
```

Comment:
My only other suggestion is that it would be useful to have a comment describing what this algorithm is looking for. Like "looks for an exact match of the service name, otherwise defaults to the first whatever, blah, blah"

Describing the intent will make the code more readable I think.